ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java

```diff
@@ -398,13 +398,14 @@ private void checkDatabase() {
                 try {
                     String createTable = sql.getProperty(CREATE_TABLE);
                     String[] statements = createTable.split(";");
-                    for (final String sql : statements) {
-                        try {
-                            statement = connection.createStatement();
-                            statement.executeUpdate(sql);
-                        } finally {
-                            safeClose(statement);
+                    try {
+                        statement = connection.createStatement();
+                        for (final String sql : statements) {
+                            statement.addBatch(sql);
                         }
+                        statement.executeBatch();
+                    } finally {
+                        safeClose(statement);
```

Comment:
The `statement` is closed twice: above at line 408, and below line 419. I think we can move the declaration of `statement` into line 400, and remove the close in line 419.